modules/tinymce/src/core/main/ts/html/FilterRegistry.ts

```diff
@@ -0,0 +1,58 @@
+import { Arr, Obj, Type } from '@ephox/katamari';
+
+import Tools from '../api/util/Tools';
+
+const explode = Tools.explode;
+
+export interface Filter<C extends Function> {
+  name: string;
+  callbacks: C[];
+}
+
+export interface FilterRegistry<C extends Function> {
+  readonly addFilter: (name: string, callback: C) => void;
+  readonly getFilters: () => Filter<C>[];
+  readonly removeFilter: (name: string, callback?: C) => void;
+}
+
+export const FilterRegistry = <C extends Function>(): FilterRegistry<C> => {
```

Comment:
I wonder what style we should use here. The constructor style here is more of a legacy in my mind from the `new Foo` style. So maybe this could be renamed `makeFilterRegistry` instead?

Reply:
I hadn't seen any indication that we'd switched to be honest, but I don't mind changing it. Though I think we can just get away with just `create` if we do that as we already have the context based on the module name.

Reply:
That would also be consistent with other more recent APIs I guess, such as Mode and Options: https://github.com/tinymce/tinymce/blob/91b891eb142d14040a12a29842338bc985284c80/modules/tinymce/src/core/main/ts/api/Mode.ts#L71

Reply:
Yeah, internally at least we should try to use more normal functions I guess then externally it might be hard since there we already have a bunch of legacy named things when it was a constructor function and some things still are classes for example.

Reply:
Done